pkg/workloadmeta/collectors/docker/docker.go

```diff
@@ -384,3 +394,18 @@ func extractNetworkIPs(networks map[string]*network.EndpointSettings) map[string
 
 	return networkIPs
 }
+
+func extractStatus(containerState *types.ContainerState) workloadmeta.ContainerStatus {
+	if containerState == nil {
+		return workloadmeta.ContainerStatusUnknown
+	}
+
+	switch containerState.Status {
+	case "removing":
+	case "exited":
+	case "dead":
+		return workloadmeta.ContainerStatusStopped
+	}
+
+	return workloadmeta.ContainerStatus(containerState.Status)
```

Comment:
I think it's safer to actually have each status in the switch above (same for the other collectors where that's not the case). There's nothing preventing any runtime from introducing another status and causing silent breakage. Also, there's no fallthrough in Go, you need to do `case "removing", "exited", "...":` ;)

Reply:
Correct, fixed!